### PR TITLE
Add guide for renaming an app

### PR DESCRIPTION
**Pull request by jessjenkins**
### What

Add a guide on how to rename an app. This is based on the Google doc created by Justin and Cat.

### How to review

Review the doc…
- Do the words make sense?
- Are there typos?
- Formatting OK? (see [the formatted page here](https://github.com/ONSdigital/dp/blob/feature/rename-app-guide/guides/RENAME_APP.md) )
- etc.

### Who can review

Anyone but me.

### CODE REVIEW

**Review by redhug1 — COMMENTED** [review]
Very minor points:

The indentation for the following lines could have two space characters removed:
13, 15, 20, 24, 31, 33, 41, 43, 45, 62

Insert a line before line 168 and change the new line 169 to:
```shell

Similarly, new line 191:
```shell

**Review by redhug1 — COMMENTED** [review]
Would it be less terse (read nicer) to say:
Issue PR to dp

as comparred to:
PR to dp

And similarly everywhere else ?

File: guides/RENAME_APP.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+- Issue PR to dp-setup:
+
+  - Add a new policy to vault-policies and dp-deployer policy
+
+  - If applicable, add routing templates to dp-setup, adding new stanzas for the new name. Leave the old ones in place
+    for now (backwards compatible). The port number should be a new one (+1 from the previous).
+
+    (The files which need changes can be found in https://github.com/ONSdigital/dp-setup#adding-a-new-app)
+
+  - If applicable, add the new port it to PORTS.md.
```
Comment (cadmiumcat) [inline guides/RENAME_APP.md:20]:
```suggestion
  - If applicable, add the new port to PORTS.md.
```

File: guides/RENAME_APP.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+  - Create a copy of the old secrets files in dp-configs for develop and name them with the NEW_SERVICE_NAME.
+
+  - Update URL route of service in the secrets.
+
+  - Update the copied manifest file to reflect the name change and repo move.
+
+    Manifest files found in https://github.com/ONSdigital/dp-configs/tree/master/manifests
+
+- Ensure all changes have been merged into develop before continuing.
```
Comment (cadmiumcat) [inline guides/RENAME_APP.md:49]: I would make this stand out
```suggestion
⚠️ Ensure all changes have been merged into develop before continuing ⚠️
```

File: guides/RENAME_APP.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+- Apply the dp-setup ansible for DEVELOP for the vault policies ONLY
+
+  STEP 4 - https://github.com/ONSdigital/dp-setup#adding-a-new-app
+
+- If the service has a port (eg. api), search for the service URL (e.g. "SEARCH_URL") in the organization in GitHub to
+  see where the service is being used and update those accordingly with NEW_SERVICE_NAME.
+
+  MAKE A NOTE OF THE PRs (CHANGES) AS THESE NEED TO BE RELEASED LATER
```
Comment (cadmiumcat) [inline guides/RENAME_APP.md:58]: and make this one stand out in a different way 😅 
```suggestion
  ⚠️ Make a note of the PRs (changes) as these need to be released later ⚠️
```

File: guides/RENAME_APP.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+- Release changes of dp-setup and check if production environment is still working as expected
+
+- Delete secrets from vault web interface production itself (Go to the secrets and click on delete)
+
+- Stop the OLD_SERVICE_NAME running in nomad production (it will be automatically removed by the garbage collector in
+  the next few days)
+
+- Update concourse to not include OLD_SERVICE_NAME (do pipeline changes
+  - https://github.com/ONSdigital/dp-ci/tree/master/pipelines/branch-manager)
```
Comment (cadmiumcat) [inline guides/RENAME_APP.md:141]: This doesn't need to be a bullet point
```suggestion
 https://github.com/ONSdigital/dp-ci/tree/master/pipelines/branch-manager)
```

File: guides/RENAME_APP.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+## Move OLD_SERVICE_NAME to NEW_SERVICE_NAME (DEVELOP)
+
+- Issue PR to dp-setup:
+
+  - Add a new policy to vault-policies and dp-deployer policy
+
+  - If applicable, add routing templates to dp-setup, adding new stanzas for the new name. Leave the old ones in place
+    for now (backwards compatible). The port number should be a new one (+1 from the previous).
+
+    (The files which need changes can be found in https://github.com/ONSdigital/dp-setup#adding-a-new-app)
```
Comment (cadmiumcat) [inline guides/RENAME_APP.md:18]:
```suggestion
    (The files which need changes can be found in the guide for [Adding a New App](https://github.com/ONSdigital/dp-setup#adding-a-new-app) )
```

File: guides/RENAME_APP.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+- Issue PR to dp-configs:
+
+  - Create a copy of the old secrets files in dp-configs for develop and name them with the NEW_SERVICE_NAME.
+
+  - Update URL route of service in the secrets.
+
+  - Update the copied manifest file to reflect the name change and repo move.
+
+    Manifest files found in https://github.com/ONSdigital/dp-configs/tree/master/manifests
```
Comment (cadmiumcat) [inline guides/RENAME_APP.md:47]:
```suggestion
  - Update the copied [manifest file](https://github.com/ONSdigital/dp-configs/tree/master/manifests) to reflect the name change and repo move.
```

File: guides/RENAME_APP.md [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+- Ensure all changes have been merged into develop before continuing.
+
+- Apply the dp-setup ansible for DEVELOP for the vault policies ONLY
+
+  STEP 4 - https://github.com/ONSdigital/dp-setup#adding-a-new-app
```
Comment (cadmiumcat) [inline guides/RENAME_APP.md:53]:
```suggestion
- Apply the dp-setup ansible for DEVELOP for the vault policies ONLY (step 4 in [Adding a New App](https://github.com/ONSdigital/dp-setup#adding-a-new-app))
```

File: guides/RENAME_APP.md [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+
+- Ship the service in develop through concourse
+
+- Update api-router config to point to new port (if the service is a backend one, e.g. API)
+
+- Update frontend-router config to point to new port (if the service is a frontend one, e.g. controller)
+
+- Apply the dp-setup ansible for DEVELOP for the consul-template
+
+  STEP 4 - https://github.com/ONSdigital/dp-setup#adding-a-new-app
```
Comment (cadmiumcat) [inline guides/RENAME_APP.md:78]:
```suggestion
- Apply the dp-setup ansible for DEVELOP for the consul-template (step 4 in [Adding a New App](https://github.com/ONSdigital/dp-setup#adding-a-new-app))
```

File: guides/RENAME_APP.md [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+- Rename secrets files in dp-configs for production AND update URL route of service in the secrets
+
+- Change branch protection in GitHub for master branch for NEW_SERVICE_NAME (this is achieved by going to settings of
+  the repo in GitHib)
+
+- Release the changes made in dp-api-router or dp-frontend-router
+
+- Apply the dp-setup ansible for PROD for the vault policies ONLY
+
+  STEP 4 - https://github.com/ONSdigital/dp-setup#adding-a-new-app
```
Comment (cadmiumcat) [inline guides/RENAME_APP.md:97]:
```suggestion
- Apply the dp-setup ansible for PROD for the vault policies ONLY (step 4 in [Adding a New App](https://github.com/ONSdigital/dp-setup#adding-a-new-app))
```

File: guides/RENAME_APP.md [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+
+- Ship the service in production through concourse
+
+- Apply the dp-setup ansible for PROD for the consul-template
+
+  STEP 4 - https://github.com/ONSdigital/dp-setup#adding-a-new-app
```
Comment (cadmiumcat) [inline guides/RENAME_APP.md:103]:
```suggestion
- Apply the dp-setup ansible for PROD for the consul-template (step 4 in [Adding a New App](https://github.com/ONSdigital/dp-setup#adding-a-new-app))
```

File: guides/RENAME_APP.md [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+
+- Issue PR to dp-configs:
+
+  - Update develop secrets of services which use OLD_SERVICE_NAME with the new port numbering
+
+- If applicable, run terraform apply (https://github.com/ONSdigital/dp-ci/tree/master/terraform)
+
+- Create a new pipeline in concourse for the newly named service.
+
+  STEP 4 - https://github.com/ONSdigital/dp-ci/tree/master/pipelines/pipeline-generator#manually-generating-the-pipelines
```
Comment (cadmiumcat) [inline guides/RENAME_APP.md:68]:
```suggestion
- Create a new pipeline in concourse for the newly named service (step 4 in [Manually Generating the Pipelines]( https://github.com/ONSdigital/dp-ci/tree/master/pipelines/pipeline-generator#manually-generating-the-pipelines))
```

File: guides/RENAME_APP.md [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+- If the service has a port (eg. api), search for the service URL (e.g. "SEARCH_URL") in the organization in GitHub to
+  see where the service is being used and update those accordingly with NEW_SERVICE_NAME.
+
+  MAKE A NOTE OF THE PRs (CHANGES) AS THESE NEED TO BE RELEASED LATER
+
+- Issue PR to dp-configs:
+
+  - Update develop secrets of services which use OLD_SERVICE_NAME with the new port numbering
+
+- If applicable, run terraform apply (https://github.com/ONSdigital/dp-ci/tree/master/terraform)
```
Comment (cadmiumcat) [inline guides/RENAME_APP.md:64]:
```suggestion
- If applicable, run terraform apply in [dp-ci](https://github.com/ONSdigital/dp-ci/tree/master/terraform)
```

File: guides/RENAME_APP.md [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+
+- Delete secrets from vault web interface production itself (Go to the secrets and click on delete)
+
+- Stop the OLD_SERVICE_NAME running in nomad production (it will be automatically removed by the garbage collector in
+  the next few days)
+
+- Update concourse to not include OLD_SERVICE_NAME (do pipeline changes
+  - https://github.com/ONSdigital/dp-ci/tree/master/pipelines/branch-manager)
+
+- Update references to OLD_SERVICE_NAME to NEW_SERVICE_NAME in dp (https://github.com/ONSdigital/dp)
```
Comment (cadmiumcat) [inline guides/RENAME_APP.md:143]:
```suggestion
- Update references to OLD_SERVICE_NAME to NEW_SERVICE_NAME in [dp](https://github.com/ONSdigital/dp)
```

**Review by cadmiumcat — COMMENTED** [review]
This looks good to me 👍  
I've suggested a couple of changes (typos), and then a lot of minor ones so that the links are clearer.

**Review by cadmiumcat — APPROVED** [review]
👍

**Pull request merged**